### PR TITLE
[Refactor] Rename http response status field to Status for compatibility

### PR DESCRIPTION
**Pull request by meegoo**
Fixes #issue

## What type of PR is this:
- [ ] BugFix
- [ ] Feature
- [ ] Enhancement
- [x] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Checklist:
- [x] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto-backported to the target branch
  - [ ] 3.1
  - [ ] 3.0
  - [ ] 2.5
  - [ ] 2.4

### CODE REVIEW

File: fe/fe-core/src/main/java/com/starrocks/http/rest/RestBaseResult.java [hunk 1]
```diff
@@ -54,6 +55,7 @@ public class RestBaseResult {
     private static final RestBaseResult OK = new RestBaseResult();
     // For compatibility, status still exists in /api/v1, removed in /api/v2 and later version.
     @Legacy
+    @SerializedName(value = "Status")
```
Comment (satanson) [inline fe/fe-core/src/main/java/com/starrocks/http/rest/RestBaseResult.java:58]: why change "status" to "Status"?

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 7 / 9 (77.78%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/http/rest/LoadAction.java|7|9|77.78%|[76, 83]|

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 7 / 9 (77.78%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/http/rest/LoadAction.java|7|9|77.78%|[76, 83]|

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 9 / 9 (100.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/http/rest/LoadAction.java|9|9|100.00%|[]|

**Review by Astralidea — APPROVED** [review]

**Review by satanson — APPROVED** [review]

**Comment by meegoo** [comment]
@mergify backport branch-3.1

**Pull request merged**